src/components/Wave/WaveEntity.ts

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2018 - 2022 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+class WaveEntity {
+  private waveLength: number[]
+  private gradientColors: string[]
+  private amplitude: number
+  private base: number
+
+  constructor(waveLength: number[], amplitude: number, base: number, gradientColors: string[]) {
+    this.waveLength = waveLength
+    this.gradientColors = gradientColors
+    this.amplitude = amplitude
+    this.base = base
+  }
+
+  public draw = (ctx: CanvasRenderingContext2D, width: number, height: number, frequency: number): void => {
```

Comment:
It would be nice to list the original source of where this code was derived, but that's me